script.js

```diff
@@ -0,0 +1,333 @@
+    this._spatialNavigator.on('focus', this._handleFocus.bind(this));
+    this._spatialNavigator.on('unfocus', this._handleUnfocus.bind(this));
+
+    // handle appendChild and removeChild in scrollbox
+    this._observer = new MutationObserver(this._handleMutation.bind(this));
+    this._observer.observe(this, {
+      childList: true
+    });
+
+    this.addEventListener('transitionend', this);
+    this.addEventListener('scroll', this);
+    this.addEventListener('focus', this);
+    this.addEventListener('blur', this);
+  };
+
+  /**
+   * Current focused element
+   */
+  Object.defineProperty(proto, 'currentElement', {
+    get: function() {
+      return this._spatialNavigator.getFocusedElement();
+    }
+  });
+
+  /**
+   * Blur the currentElement
+   */
+  proto.blur = function() {
+    if (this.currentElement) {
+      this.currentElement.blur();
+    }
+  };
+
+  /**
+   * Focus a element in scrollbox. If the input item is a number, it will focus
+   * the first navigable element in node having the same index as the number.
+   * If the item is not navigable, it will focus the first navigable element in
+   * the same node as the element.
+   */
+  proto.focus = function(item) {
+    if (!item) {
+      return this.focus(0);
+    }
+
+    if (typeof item === 'number') {
+      item = this.children[item];
+    }
+
+    // focus the first navigable item
+    if (!item.classList.contains('navigable')) {
+      item = this._getNodeFromItem(item).querySelector('.navigable');
+    }
+
+    if (item) {
+      this._spatialNavigator.focus(item);
+      return true;
+    }
+
+    return false;
+  };
+
+  /**
+   * Move the focus element
+   */
+  proto.move = function(direction) {
+    return this._spatialNavigator.move(direction);
+  };
+
+  /**
+   * Add navigable elements in scrollbox
+   */
+  proto.addNavigableElems = function(navigableElems) {
+    return this._spatialNavigator.multiAdd(navigableElems);
+  };
+
+  /**
+   * Remove navigable elements in scrollbox
+   */
+  proto.removeNavigableElems = function(navigableElems) {
+    return this._spatialNavigator.multiRemove(navigableElems);
+  };
+
+  proto.handleEvent = function (evt) {
+
+    switch (evt.type) {
+      case 'transitionend':
+        if (evt.target === this && evt.propertyName === 'transform') {
+          this._fireEvent('scrollbox-scroll', {
+            element: this
+          });
+        }
+        break;
+      case 'scroll':
+        // Gecko may scroll the scrollbox automatically, we want to prevent this
+        // behavior inorder to have correct view.
+        if (evt.target === this) {
+          evt.target.scrollLeft = 0;
+        }
+        break;
+    }
+  };
+
+  /**
+   * Get the node element of the item element
+   */
+  proto._getNodeFromItem = function(itemElem) {
+    if (!itemElem) {
+      return null;
+    }
+
+    var nodeElem = itemElem;
+    // Find the node element. If the node is removed from scrollbox, it's parent
```

Comment:
typo: `its parent`

May you try to improve the comment since it described more on the corner case, but didn't explain relations about node element and item element.